ios/FluentUI/TextField/FluentTextField.swift

```diff
@@ -0,0 +1,307 @@
+    @objc public var onDidBeginEditing: ((FluentTextField) -> Void)?
+
+    /// The closure for the action to be called in `textFieldDidEndEditing`.
+    @objc public var onDidEndEditing: ((FluentTextField) -> Void)?
+
+    /// The closure for the action to be called in `textFieldShouldReturn`. The return value of `onReturn`
+    /// will be returned in `textFieldShouldReturn`.
+    @objc public var onReturn: ((FluentTextField) -> Bool)?
+
+    /// The `FluentTextInputError` containing the `localizedDescription` that will be
+    /// displayed in the assistive text label.
+    @objc public var error: FluentTextInputError? {
+        didSet {
+            updateState()
+            updateAssistiveText()
+        }
+    }
+
+    // Hierarchy:
+    //
+    // imageTextStack
+    // |--leadingImageContainerView
+    // |--|--leadingImageView
+    // |--textStack
+    // |--|--label
+    // |--|--textfield
+    // |--|--separator
+    // |--|--assistiveTextLabel
+    @objc public init() {
+        super.init(frame: .zero)
+        textfield.validateInputText = editingChanged
+        textfield.delegate = self
+        textfield.addTarget(self, action: #selector(editingChanged), for: .editingChanged)
+
+        let textStack = UIStackView(arrangedSubviews: [label, textfield, separator, assistiveTextLabel])
+        textStack.axis = .vertical
+        textStack.alignment = .leading
+        textStack.spacing = TextFieldTokenSet.labelInputTextSpacing()
+        textStack.setCustomSpacing(4, after: separator)
+        textStack.layoutMargins = UIEdgeInsets(top: TextFieldTokenSet.topPadding(),
+                                               left: 0,
+                                               bottom: TextFieldTokenSet.bottomPadding(),
+                                               right: 0)
+        textStack.isLayoutMarginsRelativeArrangement = true
+
+        leadingImageContainerView.addSubview(leadingImageView)
+
+        let imageTextStack = UIStackView(arrangedSubviews: [leadingImageContainerView, textStack])
+        imageTextStack.axis = .horizontal
+        imageTextStack.spacing = TextFieldTokenSet.leadingIconInputTextSpacing()
+        imageTextStack.distribution = .fill
+        imageTextStack.translatesAutoresizingMaskIntoConstraints = false
+        imageTextStack.alignment = .center
+
+        let leftInset: CGFloat
+        let rightInset: CGFloat
+        if effectiveUserInterfaceLayoutDirection == .leftToRight {
+            leftInset = TextFieldTokenSet.horizontalPadding()
+            rightInset = 0
+        } else {
+            leftInset = 0
+            rightInset = TextFieldTokenSet.horizontalPadding()
+        }
+        imageTextStack.layoutMargins = UIEdgeInsets(top: 0, left: leftInset, bottom: 0, right: rightInset)
+        imageTextStack.isLayoutMarginsRelativeArrangement = true
+
+        addSubview(imageTextStack)
+
+        let safeArea = safeAreaLayoutGuide
+        NSLayoutConstraint.activate([
```

Comment:
nit: I'd suggest putting constraints logic in a private function for better readability.

Reply:
A few of the constraints could be moved out, but since most of the constraints are for the stack views I can't move them out of this method. Since I'm not saving the stacks outside of the init, I don't have any references to them, and I don't think they're worth keeping around.